dev/wlp.lib.extract/src/wlp/lib/extract/SelfExtract.java

```diff
@@ -167,6 +175,15 @@ public static int doMain(String[] args) {
             } else {
                 outputDirFromUser = new File(targetString.trim());
             }
+
+            if (outputDirFromUser.getPath().charAt(0) == '~') {
+                String pathName = outputDirFromUser.getPath();
+                String pathNameToReturn = pathName.substring(1);
+                String home = System.getenv("HOME");
+                pathNameToReturn = home + pathNameToReturn;
```

Comment:
We can create File directly with the parent path

outputDirFromUser = new File(home, pathNameToReturn);